### ISSUE TITLE
Forgetting devices corrupts knownSensors.json

### ISSUE THREAD

**Issue by AlCalzone** [node-enocean/node-enocean#33]
After forgetting a known sensor, knownSensors.json becomes corrupted. Steps to reproduce:
1. `rm knownSensors.json`
2. start the driver. knownSensors.json now contains `{}`
3. learn a device. The file now contains
```
{
    "fefcce76": {
        "id": "fefcce76",
        "eep": "f6-02-03",
        "manufacturer": "unknown",
        "title": "New RPS Switch",
        "desc": "I'm a new sensor...",
        "eepFunc": "Rocker Switch, 2 Rocker",
        "eepType": "Light Control - Application Style 1"
    }
}
```

4. forget the device. Now the file contains
```
{}    "fefcce76": {
        "id": "fefcce76",
        "eep": "f6-02-03",
        "manufacturer": "unknown",
        "title": "New RPS Switch",
        "desc": "I'm a new sensor...",
        "eepFunc": "Rocker Switch, 2 Rocker",
        "eepType": "Light Control - Application Style 1"
    }
}
```
Notice how the old contents don't get deleted.

5. restarting the driver now fails with 
```
SyntaxError: /opt/iobroker/node_modules/node-enocean/modules/knownSensors.json: Unexpected string in JSON at position 6
at JSON.parse ()
```

**Comment by Holger-Will** [comment]
strange... i'll g´have a look into this. thanks for letting me know.

**Comment by Holger-Will** [comment]
at first try i can not reproduce this behaviour... can you post a minimal testcase producing the problem?

**Comment by AlCalzone** [comment]
I've been using the [enocean adapter](https://github.com/Jey-Cee/ioBroker.enocean) for [ioBroker](https://github.com/ioBroker/ioBroker) on a Raspberry Pi, which is a bit more than "minimal". I'll try to reproduce it tomorrow without all the extras and let you know.

**Comment by AlCalzone** [comment]
So, here's a minimal test scenario: 
1. clone `https://github.com/AlCalzone/node-enocean-bug`
1. npm install
1. change the comName to the correct path to the device
1. run the app `node index.js`
  a. when it shows "learning for 15 seconds", push a button on the switch/sensor
  b. when it shows "forgetting for 15 seconds", push it again.
  c. knownSensors.json should now look like this: `{}`
1. repeat. 
1. now knownSensors.json is corrupted.

**Comment by Holger-Will** [comment]
spot on! fixed in 3.2.0 thanks!

**Comment by AlCalzone** [comment]
Great! Any idea when 3.2.0 will be published, so I can update my project?
